src/main/java/picard/illumina/IlluminaBasecallsToFastq.java

```diff
@@ -281,6 +256,12 @@ private void initialize() {
                 .withIgnoreUnexpectedBarcodes(IGNORE_UNEXPECTED_BARCODES)
                 .withBclQualityEvaluationStrategy(bclQualityEvaluationStrategy);
 
+        if(BARCODE_EXTRACT && demultiplex){
```

Comment:
I think probably `BARCODE_EXTRACT` should come down into this class (because it doesn't make sense for ExtractIlluminaBarcodes` and be renamed to something like `MATCH_BARCODES_ON_THE_FLY` (better names welcome).